### PR TITLE
start telegraf only when fluent tcp listener up and running

### PR DESCRIPTION
**Pull request by ganga1980**
This pull request introduces several changes related to the configuration and testing of Fluent Bit and Telegraf in a Windows environment, specifically within the Kubernetes context. The most significant changes include the addition of a new configuration file, a new function to test the Fluent Bit TCP listener, and updates to the `Start-Fluent-Telegraf` function.

Configuration changes:

* [`build/windows/installer/conf/fluent-bit-common.conf`](diffhunk://#diff-f18e7998da06f03365d32645a0b5021db07cfabfdcd731fb645247c913ff3c81R67-R68): Included an additional configuration file `fluent-bit-telegraf-tcp.conf`.
* [`build/windows/installer/conf/fluent-bit-telegraf-tcp.conf`](diffhunk://#diff-3f57d11679dcc7f1debb19f1b3fda289d001e61c7f71f72f895c44fd9170b9f6R1-R9): Added a new configuration file for Fluent Bit to listen to TCP inputs.
* [`kubernetes/windows/Dockerfile`](diffhunk://#diff-5c511ea7097598edb1ac655fb776c2c5db7f9875c4a853f608bcc44c6145ec64R81): Added the new `fluent-bit-telegraf-tcp.conf` file to the Dockerfile.
* [`test/fluent-bit-windows/fluent-bit-windows.yaml`](diffhunk://#diff-e63b22d944178b611c812a37b7d5d5d89ba7e6c59f66ff0478acc91a08488c3dR1-R94): Added a new DaemonSet configuration for Fluent Bit on Windows nodes in a Kubernetes cluster.

Testing and function updates:

* [`kubernetes/windows/main.ps1`](diffhunk://#diff-2da2290d6fae65f77898d02932989c6f205d597ff0641c902610425a4d8845bdR38-R58): Added a new function `Test-FluentbitTcpListener` to test if the Fluent Bit TCP listener is running.
* [`kubernetes/windows/main.ps1`](diffhunk://#diff-2da2290d6fae65f77898d02932989c6f205d597ff0641c902610425a4d8845bdR719-R724): Updated the `Start-Fluent-Telegraf` function to clear the content of the `fluent-bit-telegraf-tcp.conf` file if `TELEMETRY_CUSTOM_PROM_MONITOR_PODS` is not set or set to false.
* [`kubernetes/windows/main.ps1`](diffhunk://#diff-2da2290d6fae65f77898d02932989c6f205d597ff0641c902610425a4d8845bdL723): Moved the "Starting telegraf..." log statement to only print when `TELEMETRY_CUSTOM_PROM_MONITOR_PODS` is set to true [[1]](diffhunk://#diff-2da2290d6fae65f77898d02932989c6f205d597ff0641c902610425a4d8845bdL723) [[2]](diffhunk://#diff-2da2290d6fae65f77898d02932989c6f205d597ff0641c902610425a4d8845bdR765).
* [`kubernetes/windows/main.ps1`](diffhunk://#diff-2da2290d6fae65f77898d02932989c6f205d597ff0641c902610425a4d8845bdR802-R803): Updated the `Start-Telegraf` function to test the Fluent Bit TCP listener before starting Telegraf, and to not start Telegraf if the listener is not running [[1]](diffhunk://#diff-2da2290d6fae65f77898d02932989c6f205d597ff0641c902610425a4d8845bdR802-R803) [[2]](diffhunk://#diff-2da2290d6fae65f77898d02932989c6f205d597ff0641c902610425a4d8845bdR836-R838).

### CODE REVIEW

File: build/windows/installer/conf/fluent-bit-telegraf-tcp.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[INPUT]
```
Comment (pfrcks) [inline build/windows/installer/conf/fluent-bit-telegraf-tcp.conf:1]: I think you forgot to remove this from the common conf file

Comment (ganga1980) [reply build/windows/installer/conf/fluent-bit-telegraf-tcp.conf:1]: > I think you forgot to remove this from the common conf file

yep, just fixed.

File: kubernetes/windows/main.ps1 [hunk 2]
```diff
@@ -695,6 +716,12 @@ function Start-Fluent-Telegraf {
 
     $containerRuntime = Get-ContainerRuntime
 
+    $monitorKubernetesPods = [System.Environment]::GetEnvironmentVariable('TELEMETRY_CUSTOM_PROM_MONITOR_PODS')
```
Comment (pfrcks) [inline kubernetes/windows/main.ps1:730]: isn't this also already happening below as well when we check SIDECAR_SCRAPING_ENABLED

Comment (ganga1980) [reply kubernetes/windows/main.ps1:730]: > isn't this also already happening below as well when we check SIDECAR_SCRAPING_ENABLED
 didnt understand your comment and lets chat. SIDECAR_SCRAPING_ENABLED being set to true by default in toggle yaml(aks) and for non-aks, values.yaml to true and hence its no-op.

File: kubernetes/windows/main.ps1 [hunk 3]
```diff
@@ -35,6 +35,27 @@ function Remove-WindowsServiceIfItExists($name) {
     }
 }
 
+function Test-FluentbitTcpListener {
+    param (
+        [int]$port
+    )
+    $retryCount = 30
```
Comment (pfrcks) [inline kubernetes/windows/main.ps1]: does this mean now there can be a 30 second delay?

Comment (ganga1980) [reply kubernetes/windows/main.ps1]: > does this mean now there can be a 30 second delay?

update code to be consistent with linux agent.

**Review by pfrcks — APPROVED** [review]

**Pull request merged**